paymentsheet/src/main/java/com/stripe/android/link/ui/signup/SignUpScreenState.kt

```diff
@@ -10,5 +11,15 @@ internal data class SignUpScreenState(
     val requiresNameCollection: Boolean,
     val showKeyboardOnOpen: Boolean,
     val signUpState: SignUpState = SignUpState.InputtingPrimaryField,
+    val submitState: SubmitState = SubmitState.Idle,
     val errorMessage: ResolvableString? = null
-)
+) {
+    val canEditForm: Boolean
+        get() = submitState == SubmitState.Idle
+
+    enum class SubmitState {
+        Idle,
+        Submitting,
+        Success
+    }
```

Comment:
The above comment likely means that we can get away with a simple `isSubmitting` boolean in the state?